### PR TITLE
Need a better test client :P hah

### PR DESCRIPTION
**Pull request by willwh**
This is a start on converting this to use guzzle, although, my test client (index.php) is laughable.
I do make a successful request to encoding.com and get a 200 response from the api:

```
<?xml version="1.0"?>
<response><errors><error>No XML!</error></errors></response>
```

I'll try and dig in and make the client actually do something to properly test this over the weekend, but thought I'd throw this up so you can have a look.

### CODE REVIEW

File: lib/encoding/client.php [hunk 1]
```diff
@@ -131,44 +137,39 @@ public function sendRequest($method) {
...
+          $xml = $this->request->asXML();
+
+          $response = $client->request('POST', $this->api_url, [
+            'body' => urlencode(($xml))
+          ]);
+        }
+        elseif ($method == Encoding_Client::GET) {
+          $response = $client->request($method, $this->api_url);
+        }
+        var_dump($response->getBody()->getContents());
```
Comment (eojthebrave) [inline lib/encoding/client.php:165]: This looks like it's probably leftover debugging code.

File: lib/encoding/client.php [hunk 2]
```diff
@@ -131,44 +137,39 @@ public function sendRequest($method) {
...
+        return new Encoding_Response($response->getStatusCode(), $response->getBody());
       }
-      elseif ($method == Encoding_Client::GET) {
-        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+      catch (GuzzleHttp\Exception\ClientException $e) {
+        $response = $e->getResponse();
+        $responseBodyAsString = $response->getBody()->getContents();
+        print_r('throw a fucking error: ' . $responseBodyAsString);
```
Comment (eojthebrave) [inline lib/encoding/client.php:171]: Also, probably debugging code. :)

**Comment by eojthebrave** [comment]
:+1: Yes. I like this idea!

I have very little experience with Guzzle, but I've seen a few other projects that use the Guzzle web services client, and I wonder if that might be a good fit for this library too. http://guzzle3.readthedocs.org/webservice-client/webservice-client.html

I would probably require a pretty significant re-architecting of this library, but probably worth at least exploring. Here's an example https://github.com/intercom/intercom-php

It seems like going this route means we can mostly just provide some configuration and a basic data model, and Guzzle will take care of the heavy lifting.

Thoughts?

**Comment by willwh** [comment]
Oh, I love this idea :)

I think the doc link you provided is for Guzzle 3.x which is EOL. (One of the frustrating things about the project is the docs returned via google searches are often old).

I assume with 6.x we'll build a handler, or something.
http://docs.guzzlephp.org/en/latest/handlers-and-middleware.html?highlight=client

Yay learning! :)

**Comment by eojthebrave** [comment]
It looks like maybe that functionality was moved to https://github.com/guzzle/guzzle-services for the 6.x version? Worth looking at least.

**Comment by eojthebrave** [comment]
Or this maybe? https://github.com/guzzle/command - Though, it also looks like that feature is pretty new for Guzzle 6, and requires this PR https://github.com/guzzle/command/pull/30

**Comment by willwh** [comment]
Inclined to say, we should go this route: https://github.com/guzzle/command, it sounds excellent :)

**Pull request closed without merging**